clients/drcachesim/tracer/tracer.cpp

```diff
@@ -247,7 +247,10 @@ circular_buf_enqueue(void *data, ssize_t size, thread_id_t tid)
     do {
         /* quick racy check without locking */
         while (((queue_put + 1) & queue_size_mask) == queue_get) {
-            /* XXX: use futex/semaphore for better performance. */
+            /* XXX: there is a big slowdown for the online simulation
```

Comment:
I thought you said offline was slow too?  You plan to put the numbers in the issue tracker, right?